### PR TITLE
Feat: add mode in workflow template

### PR DESCRIPTION
**Pull request by FogDong**
Signed-off-by: FogDong <dongtianxin.tx@alibaba-inc.com>

### Description of your changes

add mode in workflow template

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

cc @barnettZQG

### CODE REVIEW

File: charts/vela-core/crds/core.oam.dev_workflows.yaml [hunk 1]
```diff
@@ -34,150 +34,16 @@ spec:
             type: string
           metadata:
             type: object
-          steps:
```
Comment (wonderflow) [inline charts/vela-core/crds/core.oam.dev_workflows.yaml:37]: why the steps field changed?

**Review by wonderflow — COMMENTED** [review]
why do we need to add mode to workflow CRD?

File: charts/vela-core/crds/core.oam.dev_workflows.yaml (hunk 1 above)
Comment (FogDong) [reply charts/vela-core/crds/core.oam.dev_workflows.yaml:37]: This field is not changed, it's just the diff displayed by github.

**Comment by FogDong** [comment]
> why do we need to add mode to workflow CRD?

When we want to run a `WorkflowRun` From `Workflow`, we want to set the mode in the `Workflow` instead of setting the mode every time when running it.

cc @barnettZQG

**Review by barnettZQG — APPROVED** [review]

**Review by Somefive — APPROVED** [review]

**Pull request merged**